daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -545,4 +610,69 @@ class IdeLedgerClient(
     userManagementStore
       .listUserRights(id, IdentityProviderId.Default)(LoggingContext.empty)
       .map(_.toOption.map(_.toList))
+
+  def getPackageIdMap(): Map[ScriptLedgerClient.ReadablePackageId, PackageId] =
+    getPackageIdPairs().toMap
+  def getPackageIdReverseMap(): Map[PackageId, ScriptLedgerClient.ReadablePackageId] =
+    getPackageIdPairs().map(_.swap).toMap
+
+  def getPackageIdPairs(): Set[(ScriptLedgerClient.ReadablePackageId, PackageId)] = {
+    originalCompiledPackages.packageIds
+      .collect(
+        Function.unlift(pkgId =>
+          for {
+            pkgSig <- originalCompiledPackages.pkgInterface.signatures.lift(pkgId)
+            meta <- pkgSig.metadata
+            readablePackageId = meta match {
+              case PackageMetadata(name, version, _) =>
+                ScriptLedgerClient.ReadablePackageId(name, version)
+            }
+          } yield (readablePackageId, pkgId)
+        )
+      )
+  }
+
+  override def vetPackages(packages: List[ScriptLedgerClient.ReadablePackageId])(implicit
+      ec: ExecutionContext,
+      esf: ExecutionSequencerFactory,
+      mat: Materializer,
+  ): Future[Unit] = {
+    val packageMap = getPackageIdMap()
+    val pkgIdsToVet = packages.map(pkg =>
+      packageMap.getOrElse(pkg, throw new IllegalArgumentException(s"Unknown package $pkg"))
+    )
+
+    unvettedPackages = unvettedPackages -- pkgIdsToVet.toSet
+    updateCompiledPackages()
+    Future.unit
+  }
```

Comment:
I think we would better wrap the body in a `Future`. Like that if the the code actually throw an exception -- which I agree should not happen -- this will not be swallowed. 

```suggestion
  ): Future[Unit] = Future{
    val packageMap = getPackageIdMap()
    val pkgIdsToVet = packages.map(pkg =>
      packageMap.getOrElse(pkg, throw new IllegalArgumentException(s"Unknown package $pkg"))
    )

    unvettedPackages = unvettedPackages -- pkgIdsToVet.toSet
    updateCompiledPackages()
  }
```